services/src/main/java/org/keycloak/email/DefaultEmailSenderProvider.java

```diff
@@ -111,7 +112,8 @@ public void send(RealmModel realm, UserModel user, String subject, String textBo
             Message msg = new MimeMessage(session);
             msg.setFrom(new InternetAddress(from));
             msg.setHeader("To", address);
-            msg.setSubject(subject);
+            msg.setSubject(MimeUtility.encodeText(subject, "utf-8", "B"));
```

Comment:
If we don't downcast `MimeMessage`, we are able to `msg.setSubject(subject, "UTF-8")`